src/libs/ajax/AzureStorage.ts

```diff
@@ -111,6 +111,40 @@ export const AzureStorage = (signal?: AbortSignal) => ({
     }), notebooks)
   },
 
+  blobMetadata: (azureStorageUrl: string) => {
+    const getObjectMetadata = async () => {
+      // assumption is made that container name guid in uri always matches the workspace Id guid it is present in
+      const workspaceId = azureStorageUrl.split('/')[3].replace('sc-', '')
```

Comment:
We might want some validation that the parsed `workspaceId` looks like a workspace ID. The object URL might point to an object in an Azure Storage Container that is not associated with Terra.

Reply:
instead of checking that workspaceId is a guid would it be reasonable to assume that if at any point getting the file via workspaceId/sas is not possible, we check to see if the file is public and if both fail display the error?

Reply:
Yeah, I think that sounds reasonable.